pkg/operator/cluster/ceph/mon/mon.go

```diff
@@ -134,6 +134,29 @@ func (c *Cluster) Start() error {
 		return fmt.Errorf("failed to initialize ceph cluster info. %+v", err)
 	}
 
+	nodes, err := c.getMonNodes()
+	if err != nil {
+		return err
+	}
+	// if there are less nodes than given monCount, fail if monCount is set, but when unset
+	// use an odd number for monCount for the nodes available
+	// zero nodes means that monCount fitted
+	if len(nodes) < c.Size {
+		return fmt.Errorf("not enough nodes available for the given monCount (%d)", c.Size)
+	}
+
+	if c.Size == 0 {
+		c.Size = len(nodes)
```

Comment:
what if we do the following in the interest of keeping it simple?
```
if c.Size == 0 {
  if len(nodes) >= defaultMonCount {
    c.Size = defaultMonCount
  } else {
    c.Size = minMonCount
  }
}
```
where minMonCount would be defined as 1.